var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -245,6 +254,10 @@ def configure_args(self):
 
         return config_args
 
+    def setup_run_environment(self, env):
+        if self.spec.satisfies("+zstd"):
+            env.append_path("HDF5_PLUGIN_PATH", self.prefix.plugins)
+
```

Comment:
1. Can `if self.spec.satisfies("+zstd")` be `if "+zstd" in self.spec`? They are the same but I would like to stick to either one or another.
2. @kgerheiser do you know whether I need to set `HDF5_PLUGIN_PATH` when the library is built with `+zstd` but I work with files that are NOT compressed with it?
3. @haampie do you think we should set the environment variable in `setup_dependent_build_environment` too? It's always confusing for me.

Reply:
1. Yes
2. No, it's not required. If you open a non-zstd compressed file the filter will not be loaded. It's all done dynamically at runtime and unless you try to read/write a zstd compressed file those code paths will not be executed. 3. Was wondering this too, though it would be the `_run_` environment.  I experimented a bit and built NCO and HDF5_PLUGIN_PATH was correctly set using a `spack load nco`.

Reply:
Thank you. Regarding `3.`, the variable might be required when a dependent package runs a build-time check of whether `zstd` is supported. Without the variable, the check might fail. There were discussions that `setup_run_environment` should be run also when building dependencies but I am not aware of the recent developments in Spack regarding this.

Reply:
Added. I could mabye see that being necessary, though I think there are better ways to check such as checking the NetCDF header for `HAVE_ZSTD`.